src/core/error/common_error_map.rs

```diff
@@ -0,0 +1,56 @@
+use de::StdError;
+
+use crate::prelude::*;
+
+pub trait MapToFailedToDeserializeJSONToValue<R> {
+    fn map_failed_to_deserialize_string<T>(
+        self,
+        input: impl AsRef<str>,
+    ) -> Result<R, CommonError>;
+    fn map_failed_to_deserialize_bytes<T>(
+        self,
+        input: &[u8],
+    ) -> Result<R, CommonError>;
+}
+
+impl<R> MapToFailedToDeserializeJSONToValue<R>
+    for Result<R, serde_json::Error>
+{
+    fn map_failed_to_deserialize_string<T>(
+        self,
+        input: impl AsRef<str>,
+    ) -> Result<R, CommonError> {
+        self.map_err(|e| {
+            error!(
+                "Failed to deserialize JSON to {}, from:\n{:?}\nError: {}",
+                type_name::<T>(),
+                input.as_ref(),
+                e
+            );
+            CommonError::FailedToDeserializeJSONToValue {
+                json_byte_count: input.as_ref().len() as u64,
+                type_name: type_name::<T>(),
+                message: format!("{}", e),
+            }
+        })
+    }
+
+    fn map_failed_to_deserialize_bytes<T>(
+        self,
+        input: &[u8],
+    ) -> Result<R, CommonError> {
+        self.map_err(|e| {
+            error!(
+                "Failed to deserialize JSON to {}, from (UTF-8):\n{:?}\nError: {}", 
+                type_name::<T>(),
+                String::from_utf8(input.to_vec()),
```

Comment:
we should probably not print the *whole* `input`, it can blow up, imagine for a 1MB Transaction (many blobs) it will blow up :) maybe do a .prefix(500) or otherwise reasonable limit?

Reply:
Is there such method on &str?

Reply:
very many different ways of doing it, you will see hundreds of suggestions when googled. Not sure if `.chars().take(n)` works if `n` > `len()`, check! if it works you can use that. (and then `.collect::<String>()`